### PR TITLE
Update/to vue3

### PR DESCRIPTION
**Pull request by saidy-moregeo**
**Changes Made So Far:**
- [ ] Upgrade to Vue 3.5.21 with compatibility packages
- [ ] Configure Vue 3 compatibility mode and build setup
- [ ] Migrate Vue Router 4 and Vuex 4 to Vue 3 APIs
- [ ] Migrate vue-i18n to v9 with legacy mode for compatibility
- [ ] Update core application components for Vue 3 compatibility (StacBrowser.vue & StacHeader.vue)
- [ ] Add TeleportPopover component using Vue 3 Teleport
- [ ] Migrate Bootstrap Vue popovers to TeleportPopover
- [ ] Convert Vue 2 filters to Vue 3 methods

**Known Issues:**
- [ ] Vue Multi-select labels not showing.
- [ ] Responsiveness of the popover items.
- [ ] Overall primary color on links.

**Future Improvements:**
- [ ] Replace vue2-datepicker with native Vue 3 datepicker
- [ ] Replace vue-multiselect with Vue 3 native alternatives
- [ ] Remove @vue/compat dependency
- [ ] Migrate from Bootstrap Vue 2 to Bootstrap Vue 3

### CODE REVIEW

File: src/components/DatePickerMixin.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+import '@vuepic/vue-datepicker/dist/main.css';
 
 export default {
   components: {
-    DatePicker: () => import('vue2-datepicker')
+    VueDatePicker
   },
   data() {
-    const dateFormat = 'YYYY-MM-DD';
+    const dateFormat = 'yyyy-MM-dd';
```
Comment (m-mohr) [inline src/components/DatePickerMixin.js:10]: If the format changed here, we also need to check whether the defined formats and imports in the locales still work.

Edit: It seems this was done, but needs further checks in the related locales files.

File: src/components/DatePickerMixin.js [hunk 2]
```diff
@@ -25,16 +27,19 @@ export default {
...
+          this.timeFormat = options.timeFormat;
+          this.dateTimeFormat = options.dateTimeFormat;
+        } catch (e) {
+          console.error(`Could not load datepicker locale for ${locale}`, e);
+          
+          this.datepickerLang = null;
         }
-        this.dateFormat = options.dateFormat;
-        this.timeFormat = options.timeFormat;
-        this.dateTimeFormat = options.dateTimeFormat;
```
Comment (m-mohr) [inline src/components/DatePickerMixin.js:37]: This needs to be tested across all locales.

File: src/components/HrefActions.vue [hunk 3]
```diff
@@ -170,7 +173,7 @@ export default {
       if (!this.isBrowserProtocol || !window.isSecureContext) {
         return false;
       }
-      else if (typeof this.data.method === 'string' && this.method.toUpperCase() !== 'GET') {
+      else if (typeof this.data.method === 'string' && this.data.method.toUpperCase() !== 'GET') {
```
Comment (m-mohr) [inline src/components/HrefActions.vue:177]: This should be backported to v4.

File: src/components/Item.vue [hunk 4]
```diff
@@ -7,14 +7,14 @@
       </b-card-title>
       <b-card-text v-if="fileFormats.length > 0 || hasDescription || isDeprecated" class="intro">
         <b-badge v-if="isDeprecated" variant="warning" class="mr-1 mt-1 deprecated">{{ $t('deprecated') }}</b-badge>
-        <b-badge v-for="format in fileFormats" :key="format" variant="secondary" class="mr-1 mt-1 fileformat">{{ format | formatMediaType }}</b-badge>
-        <template v-if="hasDescription">{{ data.properties.description | summarize }}</template>
+        <b-badge v-for="format in fileFormats" :key="format" variant="secondary" class="mr-1 mt-1 fileformat">{{ formatMediaType(format) }}</b-badge>
+        <template v-if="hasDescription">{{ summarize(data.properties.description) }}</template>
```
Comment (m-mohr) [inline src/components/Item.vue]: Probably makes sense to convert to a computed property

File: src/components/Item.vue [hunk 5]
```diff
@@ -7,14 +7,14 @@
...
       </b-card-text>
       <Keywords v-if="showKeywordsInItemCards && keywords.length > 0" :keywords="keywords" variant="primary" center />
       <b-card-text>
         <small class="text-muted">
-          <template v-if="extent">{{ extent | formatTemporalExtent }}</template>
-          <template v-else-if="data && data.properties.datetime">{{ data.properties.datetime | formatTimestamp }}</template>
+          <template v-if="extent">{{ formatTemporalExtent(extent) }}</template>
+          <template v-else-if="data && data.properties.datetime">{{ formatTimestamp(data.properties.datetime) }}</template>
           <template v-else>{{ $t('items.noTime') }}</template>
```
Comment (m-mohr) [inline src/components/Item.vue]: I guess the whole condition should be a computed property, too.

File: src/components/Item.vue [hunk 6]
```diff
@@ -112,6 +119,7 @@ export default {
     .intro {
       display: -webkit-box;
       -webkit-line-clamp: 2;
+      line-clamp: 2;
```
Comment (m-mohr) [inline src/components/Item.vue:125]: Should be backported to v4.

File: src/components/QueryableInput.vue [hunk 7]
```diff
@@ -26,6 +26,18 @@
         v-bind="validation"
       />
 
+      <VueDatePicker
+        v-if="queryable.isTemporal"
+        class="value"
+        :model-value="value?.value"
+        @update:model-value="updateValue"
+        :locale="datepickerLang"
```
Comment (m-mohr) [inline src/components/QueryableInput.vue]: We need to do this differently, because it may happen that a local that STAC Browser supports is not supported here.
We need to support this natively in STAC Browser and pass the details directly. I can help with this.

File: src/components/QueryableInput.vue [hunk 8]
```diff
@@ -109,17 +121,13 @@ export default {
   computed: {
     validation() {
       if (this.queryable.isTemporal) {
-        return {
-          type: this.queryable.isDateTime ? 'datetime' : 'date',
-          lang: this.datepickerLang,
-          format: this.queryable.isDateTime ? this.dateTimeFormat : this.dateFormat
-        };
+        return {};
       }
```
Comment (m-mohr) [inline src/components/QueryableInput.vue]: Why? The picker should still distinguish whether a date or a date and time is needed.

File: src/components/QueryableInput.vue [hunk 9]
```diff
@@ -109,17 +121,13 @@ export default {
...
       else if (this.queryable.isText) {
         return {
           type: 'text',
           minlength: this.schema.minLength,
-          maxlength: this.schema.maxLenggth,
+          maxlength: this.schema.maxLength,
```
Comment (m-mohr) [inline src/components/QueryableInput.vue:127]: Should be backported to v4.

File: src/components/SearchFilter.vue [hunk 10]
```diff
@@ -33,8 +39,22 @@
...
+            label="text"
+            :placeholder="!hasAllCollections ? $t('search.enterCollections') : $t('search.selectCollections')"
+            :tag-placeholder="$t('search.addCollections')"
+            :select-label="$t('multiselect.selectLabel')"
+            :selected-label="$t('multiselect.selectedLabel')"
+            :deselect-label="$t('multiselect.deselectLabel')"
+            :limit-text="count => $t('multiselect.andMore', {count})"
+            :loading="collectionsLoadingTimer !== null"
+            :show-no-results="false"
+            :internal-search="collections.length === 0 || !hasAllCollections"
```
Comment (m-mohr) [inline src/components/SearchFilter.vue]: I don't quite follow yet. Where are all these additional props coming from?

File: src/components/SearchFilter.vue [hunk 11]
```diff
@@ -76,11 +99,13 @@
...
             v-for="(filter, index) in filters" :key="filter.id"
-            :value.sync="filter.value"
-            :operator.sync="filter.operator"
+            :value="filter.value"
+            :operator="filter.operator"
             :queryable="filter.queryable"
             :index="index"
             :cql="cql"
+            @update:value="val => filter.value = val"
+            @update:operator="op => filter.operator = op"
```
Comment (m-mohr) [inline src/components/SearchFilter.vue]: Why not v-model:value and v-model:operator? It looks like that is the intended was of doing it?

File: src/components/SearchFilter.vue [hunk 12]
```diff
@@ -89,12 +114,15 @@
 
         <b-form-group v-if="canSort" class="sort" :label="$t('sort.title')" :label-for="ids.sort" :description="$t('search.notFullySupported')">
           <multiselect
-            :id="ids.sort" :value="sortTerm" @input="sortFieldSet"
```
Comment (m-mohr) [inline src/components/SearchFilter.vue:92]: There's a change from sortFieldSet to sortTerm directly. If that's valid, then the sortFieldSet method should be removed, right?

File: src/components/SearchFilter.vue [hunk 13]
```diff
@@ -438,6 +452,7 @@ export default {
       }
       else if (this.type === 'Global' || this.type === 'Collections') {
         let response = await stacRequest(this.$store, link);
+
```
Comment (m-mohr) [inline src/components/SearchFilter.vue:455]:
```suggestion
        
```

File: src/components/SearchFilter.vue [hunk 14]
```diff
@@ -534,17 +549,18 @@ export default {
     additionalFieldSelected(queryable) {
       const operators = queryable.getOperators(this.cql);
       this.filters.push({
+        id: `${queryable.id}-${Date.now()}-${Math.random()}`, // Unique ID
```
Comment (m-mohr) [inline src/components/SearchFilter.vue:556]: I think we do similar things in multiple cases/components. Maybe it's time to have a central instance that creates IDs in duch cases.

File: src/components/SortButtons.vue [hunk 15]
```diff
@@ -10,10 +10,11 @@
 </template>
 
 <script>
+import { defineComponent } from 'vue';
 import { BIconSortAlphaDown, BIconSortAlphaUp } from 'bootstrap-vue';
 
-export default {
-  id: "SortButtons",
+export default defineComponent({
+  name: "SortButtons",
```
Comment (m-mohr) [inline src/components/SortButtons.vue:17]: Should probably be backported to v4.

File: src/components/Source.vue [hunk 16]
```diff
@@ -1,66 +1,76 @@
...
       >
-        <b-icon-info-lg /><span class="button-label">{{ $t('source.label') }}</span>
-      </b-button>
-      <b-button
-        size="sm" variant="outline-primary" id="popover-share-btn"
-        :title="$t('source.share.withOthers')" tag="a" tabindex="0"
+        <template #trigger>
+          <b-button
+            size="sm" variant="outline-primary"
+            :title="$t('source.detailsAboutSource')" tag="a" tabindex="0"
```
Comment (m-mohr) [inline src/components/Source.vue:13]: The title changed. Is that intentional and correct? If it's a bug, then we should also backport to v4.

File: src/components/StacLink.vue [hunk 17]
```diff
@@ -99,7 +100,7 @@ export default {
       if (this.isStacBrowserLink || this.button) {
         let obj = {
           to: this.href,
-          rel: this.rel
+          rel: this.link.rel
```
Comment (m-mohr) [inline src/components/StacLink.vue:103]: This should be backported to v4.

File: src/components/StacLink.vue [hunk 18]
```diff
@@ -110,7 +111,7 @@ export default {
         const obj = {
           href: this.href,
           target: '_blank',
-          rel: this.rel,
+          rel: this.link.rel,
```
Comment (m-mohr) [inline src/components/StacLink.vue:114]: This should be backported to v4.

File: src/components/StacLink.vue [hunk 19]
```diff
@@ -134,6 +135,11 @@ export default {
         else {
           href = this.toBrowserPath(this.link.href);
         }
+        // Ensure href is a non-empty string
+        if (typeof href !== 'string' || href.length === 0) {
+          href = '/';
+        }
+        // Normalize to start with a slash for router-link navigation
```
Comment (m-mohr) [inline src/components/StacLink.vue]: In which case does this happen? We need to check whether this is correct and not Vue3-specific. If this is the case, we should also backport this to v4.

File: src/components/TeleportPopover.vue [hunk 20]
```diff
@@ -0,0 +1,382 @@
...
+  name: 'TeleportPopover',
+  props: {
+    title: {
+      type: String,
+      default: ''
+    },
+    placement: {
+      type: String,
+      default: 'bottom',
+      validator: value => ['top', 'bottom', 'left', 'right'].includes(value)
```
Comment (m-mohr) [inline src/components/TeleportPopover.vue:40]: We never use validators yet, but it's a good idea. We should probably issue a separate PR that implements validators for all components where it's useful => #672

File: src/components/auth/ApiKey.vue [hunk 21]
```diff
@@ -67,7 +67,7 @@ export default {
   },
   methods: {
     t(key) {
-      return i18n.t(key);
+  return i18n.global.t(key);
```
Comment (m-mohr) [inline src/components/auth/ApiKey.vue]:
```suggestion
     return i18n.global.t(key);
```

File: src/components/auth/Basic.vue [hunk 22]
```diff
@@ -75,7 +75,7 @@ export default {
   },
   methods: {
     t(key) {
-      return i18n.t(key);
+  return i18n.global.t(key);
```
Comment (m-mohr) [inline src/components/auth/Basic.vue]:
```suggestion
      return i18n.global.t(key);
```

File: src/components/maps/MapMixin.js [hunk 23]
```diff
@@ -53,8 +54,10 @@ export default {
       isFullScreen: false,
     };
   },
-  created() {
-    this.$root.$on('uiLanguageChanged', this.translate);
+  watch: {
+    uiLanguage() {
+      this.translate();
+    }
```
Comment (m-mohr) [inline src/components/maps/MapMixin.js:60]: I'm not sure yet whether this is actually the same, we need to check / test this. I could potentially see race conflicts happening with the code that is executed in StacBrowser.vue (that's why it has been using events in the first place).

File: src/components/maps/MapSelect.vue [hunk 24]
```diff
@@ -67,13 +69,12 @@ export default {
...
-  created() {
-    // This recreates the component so that it picks up the new translations
-    this.$root.$on('uiLanguageChanged', () => {
+    },
+    uiLanguage() {
+      () => {
       // todo: update the language resources
-    });
+    };
+    },
```
Comment (m-mohr) [inline src/components/maps/MapSelect.vue:78]: I'm not sure yet whether this is actually the same, we need to check / test this.
Also, indentation seems to be off.

File: src/components/maps/UserLocationControl.vue [hunk 25]
```diff
@@ -36,7 +36,10 @@ export default {
             const point = new Point(coords);
             view.fit(point, {maxZoom: this.maxZoom});
           },
-          error => this.$root.$emit('error', error, error.message),
+          error => this.$store.commit('showGlobalError', {
+            error,
+            message: error.message
+          }),
```
Comment (m-mohr) [inline src/components/maps/UserLocationControl.vue:42]: Note for myself: There are many cases of this in the codebase, we should clean this up a little more. I'll take a look at this later, not sure what the best way forward is yet.

File: src/i18n.js [hunk 26]
```diff
@@ -26,15 +23,18 @@ function loadLocaleConfig() {
...
 
-const i18n = new VueI18n({
+const i18n = createI18n({
+  legacy: true,
+  globalInjection: true,
   locale: CONFIG.locale,
   fallbackLocale: CONFIG.fallbackLocale,
   messages: loadLocaleConfig(),
-  // Todo: Workaround for https://github.com/kazupon/vue-i18n/issues/563
+  // Workaround for https://github.com/kazupon/vue-i18n/issues/563
```
Comment (m-mohr) [inline src/i18n.js]: Is this not a todo anymore? We should check what's the status of this in the new version

File: src/i18n.js [hunk 27]
```diff
@@ -26,15 +23,18 @@ function loadLocaleConfig() {
...
   postTranslation: (value, path) => {
     if (value === "") {
       const parts = path.split('.');
-      let message = i18n.messages[CONFIG.fallbackLocale];
+      // Access messages in a mode-agnostic way
+      let message = i18n.global.getLocaleMessage(CONFIG.fallbackLocale);
```
Comment (m-mohr) [inline src/i18n.js:38]: This needs good testing, it's an edge-cases that not happend very often, but e.g. in Japanese.

File: src/i18n.js [hunk 28]
```diff
@@ -48,7 +48,9 @@ const i18n = new VueI18n({
     return value;
   }
 });
+
 export default i18n;
+export const i18nGlobal = i18n.global;
```
Comment (m-mohr) [inline src/i18n.js]: This saves one character, do we really need it?

File: src/init.js [hunk 29]
```diff
@@ -1,12 +1,65 @@
...
+      routes: getRoutes(CONFIG),
+      scrollBehavior: (to, from, savedPosition) => {
+        if (to.path !== from.path) {
+          return { x: 0, y: 0 };
+        }
+        else {
+          return savedPosition;
+        }
+      }
+    });
```
Comment (m-mohr) [inline src/init.js:31]: We should check whether the scrollBehavior workaround is still needed in the new vue-router.
See also https://router.vuejs.org/guide/migration/#scrollBehavior-changes

File: src/locales/en-GB/datepicker.js [hunk 30]
```diff
@@ -1,6 +1,15 @@
...
 const timeFormat = 'hh:mm:ss a';
 const dateTimeFormat = `${dateFormat} ${timeFormat}`;
-const locale = import('vue2-datepicker/locale/en');
-locale.formatLocale.firstDayOfWeek = 1;
+
+const locale = {
+    ...enGB,
+    options: {
+        ...enGB.options,
+        weekStartsOn: 1, // Monday is the first day of the week.
```
Comment (m-mohr) [inline src/locales/en-GB/datepicker.js]: With the new locales from another tool, we need to check whether the overrides are still needed or whether other overrides are now need. Need to be checked against the baseline of the old tool. Applies to all locales.

File: src/locales/en/texts.json [hunk 31]
```diff
@@ -58,7 +58,8 @@
   "cancel": "Cancel",
   "catalog": {
     "license": "License",
-    "temporalExtent": "Temporal Extent"
+    "temporalExtent": "Temporal Extent",
+    "selectDateRange": "Select date range"
```
Comment (m-mohr) [inline src/locales/en/texts.json]: There's also another selectDateRange below, I think the one below should be used instead. Any references need to be updated accordingly.

File: src/models/cql2/operators/advanced.js [hunk 32]
```diff
@@ -14,11 +14,11 @@ export class CqlLike extends CqlComparisonOperator {
   }
 
   static get longLabel() {
-    return i18n.t('search.matches');
+  return i18n.global.t('search.matches');
   }
 
   static get description() {
-    return i18n.t('search.likeOperatorDescription');
+  return i18n.global.t('search.likeOperatorDescription');
```
Comment (m-mohr) [inline src/models/cql2/operators/advanced.js]: Indentation is incorrect.

File: src/router/index.js [hunk 33]
```diff
@@ -1,7 +1,9 @@
 import Browse from '../views/Browse.vue';
 
 function getPath(route, config) {
-  let path = route.params.pathMatch;
+  let pathMatch = route.params.pathMatch;
+  // pathMatch can be an array so convert to string
+  let path = Array.isArray(pathMatch) ? pathMatch.join('/') : pathMatch;
```
Comment (m-mohr) [inline src/router/index.js:6]: Is this new? In which cases is it an array?

**Comment by m-mohr** [comment]
Some general checks that we need to do based on the migration guides:

- [ ] https://router.vuejs.org/guide/migration/#The-currentRoute-property-is-now-a-ref-
- [ ] ...

File: src/components/HrefActions.vue (hunk 3 above)
Comment (saidy-moregeo) [reply src/components/HrefActions.vue:177]: Done.

File: src/components/Item.vue (hunk 6 above)
Comment (saidy-moregeo) [reply src/components/Item.vue:125]: Done.

File: src/router/index.js [hunk 34]
```diff
@@ -18,12 +20,15 @@ function getRoutes(config) {
       component: () => import("../views/SelectDataSource.vue")
     });
     routes.push({
-      path: "/search/external/(.*)",
+      path: "/search/external/:pathMatch(.*)*",
```
Comment (m-mohr) [inline src/router/index.js:26]: The migration proposes a different solution:

> `/foo/(.*)` becomes `/foo/:_(.*)`

see https://router.vuejs.org/guide/migration/#Removal-of-unnamed-parameters

Could this avoid the additional array checks?

File: src/components/QueryableInput.vue (hunk 9 above)
Comment (saidy-moregeo) [reply src/components/QueryableInput.vue:127]: Done.

File: src/components/SearchFilter.vue (hunk 11 above)
Comment (saidy-moregeo) [reply src/components/SearchFilter.vue]: Yes true, but the linter was throwing errors with the v-model - v-model directives require no argument due to Vue 2 compatibility so I tried to find a way with the separate props and event handlers.

File: src/components/SortButtons.vue (hunk 15 above)
Comment (saidy-moregeo) [reply src/components/SortButtons.vue:17]: Done.

File: src/i18n.js (hunk 28 above)
Comment (saidy-moregeo) [reply src/i18n.js]: Nope that was supposed to be removed, it's an oversight. I will remove it and make the necessary changes.

File: src/components/StacLink.vue (hunk 17 above)
Comment (saidy-moregeo) [reply src/components/StacLink.vue:103]: Done.

File: src/components/StacLink.vue (hunk 18 above)
Comment (saidy-moregeo) [reply src/components/StacLink.vue:114]: Done.

File: src/views/SelectDataSource.vue [hunk 35]
```diff
@@ -12,15 +12,17 @@
     <hr v-if="stacIndex.length > 0">
     <b-form-group v-if="stacIndex.length > 0" class="stac-index">
       <template #label>
-        <i18n path="index.selectStacIndex">
-          <template #stacIndex>
-            <a href="https://stacindex.org" target="_blank">STAC Index</a>
-          </template>
-        </i18n>
+        <span v-html="$t('index.selectStacIndex', { stacIndex: '<a href=&quot;https://stacindex.org&quot; target=&quot;_blank&quot;>STAC Index</a>' })" />
```
Comment (m-mohr) [inline src/views/SelectDataSource.vue]: This new veriant is not really clean looking. We should check for better alternatives.

File: src/views/SelectDataSource.vue [hunk 36]
```diff
@@ -35,20 +37,23 @@
 </template>
 
 <script>
-import { BForm, BFormGroup, BFormInput, BListGroup, BListGroupItem } from 'bootstrap-vue';
+import { BForm, BFormGroup, BFormInput, BListGroup, BListGroupItem, BBadge, BButton } from 'bootstrap-vue';
```
Comment (m-mohr) [inline src/views/SelectDataSource.vue]: The new imports are weird, isn't that done by the Plugin system of Bootstrap? Or do we generally switch the behaviour here?

File: vue.config.js [hunk 37]
```diff
@@ -34,10 +34,16 @@ const vueConfig = {
...
   chainWebpack: webpackConfig => {
+    // Configure Vue 3 compatibility mode
+    webpackConfig.resolve.alias.set('vue', '@vue/compat');
+
     webpackConfig.plugin('define').tap(args => {
       args[0].STAC_BROWSER_VERSION = JSON.stringify(pkgFile.version);
       args[0].CONFIG_PATH = JSON.stringify(configFile);
       args[0].CONFIG_CLI = JSON.stringify(argv);
+      args[0].__VUE_OPTIONS_API__ = true;
+      args[0].__VUE_PROD_DEVTOOLS__ = false;
```
Comment (m-mohr) [inline vue.config.js]: That's the default value and is not needed.

```suggestion
```

File: vue.config.js [hunk 38]
```diff
@@ -34,10 +34,16 @@ const vueConfig = {
...
+      args[0].__VUE_PROD_HYDRATION_MISMATCH_DETAILS__ = false;
```
Comment (m-mohr) [inline vue.config.js]: That's the default value and is not needed.
```suggestion
```

File: src/components/SearchFilter.vue (hunk 11 above)
Comment (m-mohr) [reply src/components/SearchFilter.vue]: We need to check whether the linter actually works correctly in v3.

File: src/components/Item.vue (hunk 5 above)
Comment (saidy-moregeo) [reply src/components/Item.vue]: Done.

File: src/components/Item.vue (hunk 4 above)
Comment (saidy-moregeo) [reply src/components/Item.vue]: Done.

File: src/components/SearchFilter.vue (hunk 10 above)
Comment (saidy-moregeo) [reply src/components/SearchFilter.vue]: The previous code used v-bind="collectionSelectOptions" which was a computed property that contained all those props in one object, I was facing some issue with the multiple select then I decided to go with the inline approach.

I can revert back to using a computed property and find a solution to the issue.

Comment (m-mohr) [reply src/components/SearchFilter.vue]: That would be a little cleaner, I think. Trying to avoid too much logic in the template. What kind of issue?

Comment (saidy-moregeo) [reply src/components/SearchFilter.vue]: The selected values were not set or shown.

File: src/components/StacLink.vue (hunk 19 above)
Comment (saidy-moregeo) [reply src/components/StacLink.vue]: Nope it's not related to Vue 3, just a safeguard I added while debugging a bug I was facing with he navigation - to handle scenarios when the Stac data is malformed (missing required fields needed to generate the browser path might lead to getBrowserPath() returning empty or null) or invalid Link href eg. href = ''/null/undefined  that toBrowserPath can't process.

File: src/components/SearchFilter.vue [hunk 39]
```diff
@@ -225,31 +240,32 @@ export default {
...
         loading: this.collectionsLoadingTimer !== null,
         showNoResults: false,
-        internalSearch: !isResult
+        internalSearch: !isResult,
       };
     },
-    collectionSearchLink() {
-      return this.parent instanceof CatalogLike && this.parent.getApiCollectionsLink();
-    },
     canSearchCollectionsFreeText() {
```
Comment (m-mohr) [inline src/components/SearchFilter.vue]: There's quite a bit of noise in this file diff. Can we revert to the original formatting as much as possible to ensure a clean history?

File: src/components/SearchFilter.vue (hunk 11 above)
Comment (saidy-moregeo) [reply src/components/SearchFilter.vue]: Done.

File: src/components/SearchFilter.vue (hunk 12 above)
Comment (saidy-moregeo) [reply src/components/SearchFilter.vue:92]: Yes it's valid,  Vue 3 v-model handles the two-way data binding so we don't need a setter method.

File: src/components/Source.vue (hunk 16 above)
Comment (saidy-moregeo) [reply src/components/Source.vue:13]: I don't follow, title has changed there are two teleport popovers  one for source and another for share and the trigger buttons still maintains the same titles.

File: src/i18n.js (hunk 26 above)
Comment (saidy-moregeo) [reply src/i18n.js]: Actually it's an oversight the workaround for the Crowdin empty string issue is still relevant and needs checking especially if we move completely from the i18n legacy mode to using the recommended Composition API we would need alternative solution as the postTranslation won't be available.

File: src/init.js (hunk 29 above)
Comment (saidy-moregeo) [reply src/init.js:31]: I mean I was missing the breaking changes of scrollToOptions left and top, but we still the scroll behavior management as it's not automatically handled correctly by Vue. I have added the missing changes.

File: src/locales/en/texts.json (hunk 31 above)
Comment (saidy-moregeo) [reply src/locales/en/texts.json]: Actually that's the one being used, I will remove this one on the catalog.

File: src/router/index.js (hunk 34 above)
Comment (saidy-moregeo) [reply src/router/index.js:26]: I don't follow do we have any unnamed parameters and regarding the array checks is because Vue Router v4's (.*)* pattern returns arrays by design so we need to handle the split, this can be avoided by removing the trailing * but it's the recommended approach along with the catch-all parameter.

File: vue.config.js (hunk 37 above)
Comment (saidy-moregeo) [reply vue.config.js]: True these are the default values, but based on the documentation it's recommended to set them up.

https://vuejs.org/api/compile-time-flags.html#:~:text=Vue%20will%20work%20even%20if%20these%20flags%20are%20not%20explicitly%20configured.%20However%2C%20it%20is%20recommended%20to%20always%20configure%20them%20so%20that%20the%20relevant%20features%20can%20be%20properly%20removed%20when%20possible.

File: vue.config.js (hunk 38 above)
Comment (saidy-moregeo) [reply vue.config.js]: https://vuejs.org/api/compile-time-flags.html#:~:text=Vue%20will%20work%20even%20if%20these%20flags%20are%20not%20explicitly%20configured.%20However%2C%20it%20is%20recommended%20to%20always%20configure%20them%20so%20that%20the%20relevant%20features%20can%20be%20properly%20removed%20when%20possible.

File: src/components/SearchFilter.vue (hunk 39 above)
Comment (saidy-moregeo) [reply src/components/SearchFilter.vue]: Okay noted will do.

File: src/views/SelectDataSource.vue (hunk 35 above)
Comment (saidy-moregeo) [reply src/views/SelectDataSource.vue]: Okay noted.

File: src/views/SelectDataSource.vue (hunk 36 above)
Comment (saidy-moregeo) [reply src/views/SelectDataSource.vue]: Yes they are available globally via the plugin system, I guess was carried away with the fact that some of the others are also globally available, but still imported. Will remove it.

File: vue.config.js (hunk 37 above)
Comment (m-mohr) [reply vue.config.js]: I think that's not what they mean there. I think they mean you should set them if you don't need any functionality, e.g. remove options api. It would still work if you keep it, but lead to bigger bundle size. But just specifying the default value again doesn't make any difference.

File: vue.config.js (hunk 38 above)
Comment (m-mohr) [reply vue.config.js]: See above

File: .eslintrc [hunk 40]
```diff
@@ -11,9 +11,14 @@
...
   },
   "extends": [
-    "plugin:vue/strongly-recommended",
+    "plugin:vue/vue3-strongly-recommended",
     "eslint:recommended"
   ],
+  "env": {
+    "browser": true,
+    "es2022": true,
+    "node": true
```
Comment (m-mohr) [inline .eslintrc]: Why node?

File: src/components/Source.vue (hunk 16 above)
Comment (m-mohr) [reply src/components/Source.vue:13]: The diff was confusing, sorry. Seems all correct after further inspection.

File: src/components/StacLink.vue (hunk 19 above)
Comment (m-mohr) [reply src/components/StacLink.vue]: Which STAC resource lead to this? I'd like to check in which circumstances this appears.

File: src/i18n.js (hunk 27 above)
Comment (m-mohr) [reply src/i18n.js:38]: Seems to work fine.

File: src/components/maps/MapSelect.vue (hunk 24 above)
Comment (m-mohr) [reply src/components/maps/MapSelect.vue:78]: Let's replace the old EventBus with mitt (as proposed in https://v3-migration.vuejs.org/breaking-changes/events-api.html#event-bus) if we don't find an easy other solution.

File: src/router/index.js (hunk 34 above)
Comment (m-mohr) [reply src/router/index.js:26]: I checked it, you are right. The conversion is correct.

File: src/components/QueryableInput.vue (hunk 8 above)
Comment (saidy-moregeo) [reply src/components/QueryableInput.vue]: With the new date picker this can be explicitly added as a prop.

File: src/locales/en-GB/datepicker.js (hunk 30 above)
Comment (saidy-moregeo) [reply src/locales/en-GB/datepicker.js]: We don't need the overrides with the date-fns.

Comment (m-mohr) [reply src/locales/en-GB/datepicker.js]: Did you check this for all locales?

File: .eslintrc (hunk 40 above)
Comment (saidy-moregeo) [reply .eslintrc]: It was to handle any node scripts, but it's not necessary due the fact the only place it's needed, it's on the config files and that's ignored by the linter.

File: src/components/StacLink.vue (hunk 19 above)
Comment (saidy-moregeo) [reply src/components/StacLink.vue]: I wasn't experiencing on a specific STAC, but while I was upgrading to the new Vue router I was facing some issues that related to the Array check and while debugging I added this check to make sure it handles STAC data that's malformed I can remove it if you think is an overkill.

File: src/locales/en-GB/datepicker.js (hunk 30 above)
Comment (saidy-moregeo) [reply src/locales/en-GB/datepicker.js]: Not really only the once we had a override for the first day of the week.

Comment (m-mohr) [reply src/locales/en-GB/datepicker.js]: Okay, let's keep this task open for later.

Comment (m-mohr) [reply src/locales/en-GB/datepicker.js]: Added to the list in the initial comment, so that we don't forget and resolve here.

File: src/components/StacLink.vue (hunk 19 above)
Comment (m-mohr) [reply src/components/StacLink.vue]: Yeah, I think I'd remove it. If needed, we could add it to the toBrowserPath function so that it applies everywhere, but just checking it in one place seems like a potential issue.

Comment (m-mohr) [reply src/components/StacLink.vue]: Done.
